server/errors/encoded_errors_test.go

```diff
@@ -29,6 +29,69 @@ func TestEncodedError(t *testing.T) {
 			expectedCodeString:     "20",
 			isValidEncodedError:    false,
 		},
+		{
+			_name:                  "auth error: invalid subject",
+			errorCode:              int32(openfgapb.AuthErrorCode_auth_failed_invalid_subject),
+			message:                "error message",
+			expectedHTTPStatusCode: http.StatusUnauthorized,
+			expectedCode:           1001,
```

Comment:
What grpc status code do these get mapped to again? We should probably be asserting that behavior as well.

Reply:
Isn't the idea that these get mapped to the code specified in `errorCode`?

Reply:
But we map the code to a canonical HTTP status code, so why don't we do the same for the grpc protocol? It feels weird to me that we deviate from the grpc status code but map to HTTP. If a grpc client wants to integrate with our API, just like an HTTP client would, then they should have error status codes that make sense according to the protocol they are using.

Reply:
In fact, we convert standard gRPC status codes into our own: https://github.com/openfga/openfga/blob/78045ede7707a4e27c74b1856385c0808d05944c/server/errors/encoded_errors.go#L190

Reply:
Hmm.. Since this is what we do today let's get this in. But we really need to revisit this. It doesn't make any sense to expose a customer API error code over grpc when the grpc client is going to operate on the canonical codes of the protocol it is using (e.g. grpc). Like, for example, if I am a grpc client and I get a `Unauthenticated` status code back, then I can handle it according to the standards of protocol. But if I get back a custom error code such as 1001, then as a grpc client I don't know what the hell to do with it :) .

Reply:
https://grpc.github.io/grpc/core/md_doc_statuscodes.html

So, for example, if the token has invalid claims, the gRPC status code would be `16 UNAUTHENTICATED`? Would we add the "invalid claims" as a message, too?